Index: gtk+3.0-3.4.2/gtk/gtkicontheme.c
===================================================================
--- gtk+3.0-3.4.2.orig/gtk/gtkicontheme.c	2012-05-02 08:46:50.000000000 -0400
+++ gtk+3.0-3.4.2/gtk/gtkicontheme.c	2012-05-18 10:00:12.887833518 -0400
@@ -1331,23 +1331,6 @@
   
   ensure_valid_themes (icon_theme);
 
-  /* for symbolic icons, do a search in all registered themes first;
-   * a theme that inherits them from a parent theme might provide
-   * an alternative highcolor version, but still expect the symbolic icon
-   * to show up instead.
-   */
-  if (icon_names[0] &&
-      g_str_has_suffix (icon_names[0], "-symbolic"))
-    {
-      for (l = priv->themes; l; l = l->next)
-        {
-          IconTheme *theme = l->data;
-          icon_info = theme_lookup_icon (theme, icon_names[0], size, allow_svg, use_builtin);
-          if (icon_info)
-            goto out;
-        }
-    }
-
   for (l = priv->themes; l; l = l->next)
     {
       IconTheme *theme = l->data;
