diff --git a/gtk/gtktreemodel.c b/gtk/gtktreemodel.c
index 4bbd803..a2abead 100644
--- a/gtk/gtktreemodel.c
+++ b/gtk/gtktreemodel.c
@@ -2208,8 +2208,8 @@ gtk_tree_row_reference_unref_path (GtkTreePath  *path,
  * Creates a row reference based on @path.
  *
  * This reference will keep pointing to the node pointed to
- * by @path, so long as it exists. It listens to all signals
- * emitted by @model, and updates its path appropriately. If
+ * by @path, so long as it exists. Any changes that occur on @model are
+ * propagated, and the path is updated appropriately. If
  * @path isn't a valid path in @model, then %NULL is returned.
  *
  * Return value: a newly allocated #GtkTreeRowReference, or %NULL
diff --git a/gtk/gtkmessagedialog.c b/gtk/gtkmessagedialog.c
index a6a4889..c46cda4 100644
--- a/gtk/gtkmessagedialog.c
+++ b/gtk/gtkmessagedialog.c
@@ -50,6 +50,10 @@
  * convenience widget; you could construct the equivalent of #GtkMessageDialog
  * from #GtkDialog without too much effort, but #GtkMessageDialog saves typing.
  *
+ * One difference from #GtkDialog is that #GtkMessageDialog sets the
+ * #GtkWindow:skip-taskbar-hint property to %TRUE, so that the dialog is hidden
+ * from the taskbar by default.
+ *
  * The easiest way to do a modal message dialog is to use gtk_dialog_run(), though
  * you can also pass in the %GTK_DIALOG_MODAL flag, gtk_dialog_run() automatically
  * makes the dialog modal and waits for the user to respond to it. gtk_dialog_run()
diff --git a/gtk/gtkentry.c b/gtk/gtkentry.c
index 28bf32b..f5fa92d 100644
--- a/gtk/gtkentry.c
+++ b/gtk/gtkentry.c
@@ -790,8 +790,9 @@ gtk_entry_class_init (GtkEntryClass *class)
    *
    * Sets the text area's border between the text and the frame.
    *
-   * Deprecated: 3.4: Use the standard border and padding CSS properties;
-   *   the value of this style property is ignored.
+   * Deprecated: 3.4: Use the standard border and padding CSS properties
+   *   (through objects like #GtkStyleContext and #GtkCssProvider); the value
+   *   of this style property is ignored.
    */
   g_object_class_install_property (gobject_class,
                                    PROP_INNER_BORDER,
@@ -1379,8 +1380,9 @@ gtk_entry_class_init (GtkEntryClass *class)
    *
    * Since: 2.16
    *
-   * Deprecated: 3.4: Use the standard margin CSS property;
-   *   the value of this style property is ignored.
+   * Deprecated: 3.4: Use the standard margin CSS property (through objects
+   *   like #GtkStyleContext and #GtkCssProvider); the value of this style
+   *   property is ignored.
    */
   gtk_widget_class_install_style_property (widget_class,
 					   g_param_spec_boxed ("progress-border",
@@ -1875,8 +1877,9 @@ gtk_entry_class_init (GtkEntryClass *class)
    *
    * Since: 2.10
    *
-   * Deprecated: 3.4: Use the standard border and padding CSS properties;
-   *   the value of this style property is ignored.
+   * Deprecated: 3.4: Use the standard border and padding CSS properties
+   *   (through objects like #GtkStyleContext and #GtkCssProvider); the value
+   *   of this style property is ignored.
    */
   gtk_widget_class_install_style_property (widget_class,
 					   g_param_spec_boxed ("inner-border",
@@ -7331,8 +7334,9 @@ gtk_entry_get_has_frame (GtkEntry *entry)
  *
  * Since: 2.10
  *
- * Deprecated: 3.4: Use the standard border and padding CSS properties;
- *   the value set with this function is ignored by #GtkEntry.
+ * Deprecated: 3.4: Use the standard border and padding CSS properties (through
+ *   objects like #GtkStyleContext and #GtkCssProvider); the value set with
+ *   this function is ignored by #GtkEntry.
  **/
 void
 gtk_entry_set_inner_border (GtkEntry        *entry,
@@ -7354,8 +7358,9 @@ gtk_entry_set_inner_border (GtkEntry        *entry,
  *
  * Since: 2.10
  *
- * Deprecated: 3.4: Use the standard border and padding CSS properties;
- *   the value returned by this function is ignored by #GtkEntry.
+ * Deprecated: 3.4: Use the standard border and padding CSS properties (through
+ *   objects like #GtkStyleContext and #GtkCssProvider); the value returned by
+ *   this function is ignored by #GtkEntry.
  **/
 const GtkBorder *
 gtk_entry_get_inner_border (GtkEntry *entry)
